### PR TITLE
Fix GHEC with Data Residency (`*.ghe.com`) base URL

### PR DESCRIPTION
**Pull request by shahzadhaider1**
GitHub App and token auth against GHEC with Data Residency (`https://SUBDOMAIN.ghe.com`) sent API and token-refresh requests to the wrong host, causing 401s. GHE.com serves its REST/GraphQL API at the root of `api.SUBDOMAIN.ghe.com` (like `api.github.com`), not under the GHES `/api/v3` path.

## Changes
- Detect GHE.com endpoints by `.ghe.com` host suffix (`isGHECloud`).
- Normalize GHE.com endpoints to the API subdomain `https://api.SUBDOMAIN.ghe.com/`, accepting either the web (`SUBDOMAIN.ghe.com`) or API (`api.SUBDOMAIN.ghe.com`) form.
- Build REST and GraphQL clients for GHE.com at the root level (no `/api/v3`); GraphQL at `/graphql`.
- GitHub App: set `ghinstallation` transport `BaseURL` to the GHE.com API base, and use `NewFromAppsTransport` so the API client's token refresh inherits that base instead of defaulting to `api.github.com`. Trim the trailing slash, which `ghinstallation` does not expect.

## Unchanged
- GitHub.com / GHEC standard -> `api.github.com`.
- GHES -> `WithEnterpriseURLs` (`/api/v3`).

## Tests
- `isGHECloud` detection (GHE.com vs GHES vs github.com, plus edge cases).
- GHE.com endpoint normalization (web/API form, trailing slash, port, path).
- GHE.com REST client base has no `/api/v3`; GraphQL resolves to `/graphql`.
- App transport `BaseURL` is set to the enterprise endpoint for token refresh.
- Regression checks for github.com and GHES clients.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

---

> [!NOTE]
> **Medium Risk**
> Changes authentication and API client construction for enterprise GitHub endpoints; mistakes could break GHES or github.com, but behavior is split by endpoint type with regression tests.
> 
> **Overview**
> Fixes **GitHub Enterprise Cloud with data residency** (`*.ghe.com`) so REST, GraphQL, and GitHub App token refresh hit the correct hosts instead of GHES-style `/api/v3` paths or `api.github.com`.
> 
> The connector now **detects** GHE.com by host suffix, **normalizes** user endpoints to `https://api.SUBDOMAIN.ghe.com/` (web or API form), and builds **root-level** REST/GraphQL clients like `api.github.com`. **GHES** still uses `WithEnterpriseURLs`; **github.com** is unchanged.
> 
> For **GitHub Apps**, `ghinstallation` gets a GHE.com-aware `BaseURL` via `appsBaseURL`, installation transports no longer override that base, and app clients use `createGHECloudClient` instead of enterprise URL suffixes.
> 
> **Token** sources treat GHE.com as enterprise for enumeration (same as GHES). Tests cover detection, normalization, client URLs, and app transport base URLs, with regressions for cloud and GHES.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e4febc26d22e382bcc105e9f9191960ef0f5812e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by MuneebUllahKhan222 — APPROVED** [review]

**Review by camgunz — APPROVED** [review]

File: pkg/sources/github/connector.go [hunk 1]
```diff
@@ -34,6 +36,81 @@ type Connector interface {
...
+		u.Host = host + ":" + port
+	} else {
+		u.Host = host
+	}
+
+	// go-github requires a trailing slash on BaseURL.
+	u.Path = "/"
+
+	return u.String(), nil
+}
```
[the remark this answers is not in the document]
Comment (shahzadhaider1) [reply pkg/sources/github/connector.go:91]: This one doesn't apply to this PR. `githubWebHost` lives in `urls.go`, which this PR doesn't touch, and the normalized endpoint never reaches it.
